pkg/objstore/swift/swift.go

```diff
@@ -0,0 +1,314 @@
+/*
+Copyright 2018 eBay Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package swift implements common object storage abstractions against OpenStack swift APIs.
+package swift
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"io"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/improbable-eng/thanos/pkg/objstore"
+
+	"github.com/go-kit/kit/log"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/objectstorage/v1/containers"
+	"github.com/gophercloud/gophercloud/openstack/objectstorage/v1/objects"
+	"github.com/gophercloud/gophercloud/pagination"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	// Class A operations.
+	opObjectsList  = "objects.list"
+	opObjectInsert = "object.insert"
+
+	// Class B operation.
+	opObjectGet = "object.get"
+
+	// Free operations.
+	opObjectDelete = "object.delete"
+)
+
+// DirDelim is the delimiter used to model a directory structure in an object store bucket.
+const DirDelim = "/"
+
+type swiftConfig struct {
+	AuthUrl       string `yaml:"authUrl"`
+	Username      string `yaml:"username,omitempty"`
+	UserId        string `yaml:"userId,omitempty"`
+	Password      string `yaml:"password"`
+	DomainId      string `yaml:"domainId,omitempty"`
+	DomainName    string `yaml:"domainName,omitempty"`
+	TenantID      string `yaml:"tenantId,omitempty"`
+	TenantName    string `yaml:"tenantName,omitempty"`
+	RegionName    string `yaml:"regionName,omitempty"`
+	ContainerName string `yaml:"containerName"`
```

Comment:
We need to standardise on the yaml casing ... think we should stick with what Prom is currently using so `snake_case` see new issue #596 .
@bwplotka thoughts?

Reply:
changed to `snake_case`